js-extensions/src/ExtensionRenderer.jsx

```diff
@@ -1,113 +1,111 @@
      * _renderAllExtensions.
      */
-    render: function() {
+    render() {
         var extensions = this.state.extensions;
-
         if (!extensions) {
-            // Initial state. componentDidMount will kick in and load the extensions.
-            return null;
-        } else if (extensions.length === 0) {
-            console.warn('No "' + this.props.name + '" ExtensionPoint implementations were found across the installed plugin set. See ExtensionList:');
-            console.log(store.getExtensionList());
-            return null;
-        } else {
-            // Add a <div> for each of the extensions. See the __renderAllExtensions function.
-            var extensionDivs = [];
-            for (var i = 0; i < extensions.length; i++) {
-                extensionDivs.push(<div key={i}/>);
-            }
-            return React.createElement(this.props.wrappingElement, null, extensionDivs);
+            return null; // this is called before extension data is available
         }
-    },
+        
+        // Add a <div> for each of the extensions. See the __renderAllExtensions function.
+        var extensionDivs = [];
+        for (var i = 0; i < extensions.length; i++) {
+            extensionDivs.push(<div key={i}/>);
+        }
+        return React.createElement(this.props.wrappingElement, null, extensionDivs);
+    }
 
     /**
      * For each extension, we have created a "leaf node" element in the DOM. This method creates a new react hierarchy
      * for each, and instructs it to render. From that point on we have a separation that keeps the main app insulated
      * from any plugin issues that may cause react to throw while updating. Inspired by Nylas N1.
      */
-    _renderAllExtensions: function() {
+    _renderAllExtensions() {
         // NB: This needs to be a lot cleverer if the list of extensions for a specific point can change;
         // We will need to link each extension with its containing element, in some way that doesn't leak :) Easy in
         // browsers with WeakMap, less so otherwise.
         const el = ReactDOM.findDOMNode(this);
         if (el) {
             const children = el.children;
             if (children) {
-                const extensions = store.getExtensions(this.props.name);
+                const extensions = this.state.extensions;
 
                 // The number of children should be exactly the same as the number
                 // of extensions. See the render function for where these are added.
                 if (!extensions || extensions.length !== children.length) {
-                    console.error('Unexpected error in Jenkins ExtensionPoint rendering (' + this.props.name + '). Expecting a child DOM node for each extension point.');
+                    console.error('Unexpected error in Jenkins ExtensionRenderer rendering (' + this.props.extensionPoint + '). Expecting a child DOM node for each extension point.');
```

Comment:
Ok sorry ... the type name is `ExtensionRenderer` but the react element is `Extensions.Renderer `. Hmmm.